src/main/java/uk/gov/hmcts/divorce/payment/PaymentStatusService.java

```diff
@@ -58,16 +67,38 @@ public void hasSuccessFulPayment(List<uk.gov.hmcts.reform.ccd.client.model.CaseD
                 .toList()
```

Comment:
There are a few methods that refer to `applicationPayments` and we may need to update them to use `finalOrderPayments` if the case state is AwaitingFinalOrderPayment? For example:

line 55: https://github.com/hmcts/nfdiv-case-api/pull/4048/files#diff-4dd2c466c2e3b25636dc560ad1bc058f458ecaf36b92ed027f677714cc7e753bR55
line 105: https://github.com/hmcts/nfdiv-case-api/pull/4048/files#diff-4dd2c466c2e3b25636dc560ad1bc058f458ecaf36b92ed027f677714cc7e753bR105
line 114: https://github.com/hmcts/nfdiv-case-api/pull/4048/files#diff-4dd2c466c2e3b25636dc560ad1bc058f458ecaf36b92ed027f677714cc7e753bR114

Reply:
Updated code to refer to correct payment collection